CMSgov/eAPD
# using  JWT_SECRET to sign JWT

resolves #3128

**Description-**

This pull request converts the secret used for signing the JWT to use one from the environment instead of hard coded.

- previously issued JWT's will be invalid (unless you set your secret to 'ABCDEFGHIJKLMNOPQRSTUVWXYZ123456789'
- you MUST stop your docker containers
- you MUST then set a local variable called JWT_SECRET (export JWT_SECRET=<YOUR_SECRET>) 
- you should now restart docker
- **** The environment variable is pulled into docker when it starts.  Changing it while docker runs will have no effect.

**This pull request also touches…**
nothing.  The JWT will perform exactly as before, simply have a more secure key.

- possible side effects of this change
   - This should be very isolated.  If the JWT is working, the secret must be correct. 

**This pull request was tested in the follow ways…**

see manual verification below.  

Also, unit tests were created.

**Steps to manually verify this change...**

The most direct way to verify this change is:
- log in to the application
- look at your cookies for one with a key of gov.cms.eapd.api-token
- the cookie value is url encoded like this  {%22accessToken%22:%22<TOKEN_IS_HERE>%22}
- copy token to clipboard and go to JWT.io
- add your secret to 'VERIFY SIGNATURE' 
- verify that the signature, the last section if split by periods, has not changed


### This pull request is ready to review when...


### This pull request can be merged when…
